common/cards/alter-egos/effects/turtle-shell.ts

```diff
@@ -1,10 +1,46 @@
-import {CardComponent, ObserverComponent} from '../../../components'
+import assert from 'assert'
+import {
+	CardComponent,
+	ObserverComponent,
+	PlayerComponent,
+	StatusEffectComponent,
+} from '../../../components'
 import query from '../../../components/query'
-import {GameModel} from '../../../models/game-model'
-import {beforeDefence} from '../../../types/priorities'
+import {PlayerEntity} from '../../../entities'
+import {GameModel, GameValue} from '../../../models/game-model'
+import LooseShellEffect from '../../../status-effects/loose-shell'
+import {beforeDefence, onTurnEnd} from '../../../types/priorities'
 import {attach} from '../../base/defaults'
 import {Attach} from '../../base/types'
 
+type ActiveInfo = {
+	readonly hermit: CardComponent
+	readonly stage: HermitState
+}
+
+type HermitState =
+	| 'opponent-activated'
+	| 'player-activated'
+	| 'defend-then-discard'
+	| 'too-late'
+
+type InstanceState =
+	| 'opponent-activated'
+	| 'player-activated'
+	| 'defend-then-discard'
+	| 'inactive'
+
+const lastActiveHermit = new GameValue<
+	Record<PlayerEntity, ActiveInfo | undefined>
+>(() => {
+	return {}
+})
+
+const startProtectionEntry = (hermit: CardComponent) =>
+	`$p${hermit.props.name}$ is now protected by $e${TurtleShell.name}$`
+const endProtectionEntry = (hermit: CardComponent, rowIndex: number) =>
+	`$p${hermit.props.name}$ is no longer protected by $e${TurtleShell.name} (${rowIndex + 1})$`
```

Comment:
```suggestion
	`$p${hermit.props.name}$ is no longer protected by $e${TurtleShell.name}$`
```

Reply:
I included the Shell's row number here to avoid log ordering issues, for example a player changes between two Turtle Shells and either both hermits have the same name or `startProtectionEntry` comes before `endProtectionEntry` and creates ambiguous logs.

Reply:
I actually had to move it to after the hermits name, misunderstood what this was for.